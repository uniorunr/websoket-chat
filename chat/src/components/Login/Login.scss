@import "../../colors/colors";

.login-section {
  display: flex;
  justify-content: center;
  align-items: center;
  width: fit-content;
  height: fit-content;
  border: 2px solid $black-color;
  border-radius: 10px;
  margin-bottom: 50px;

  .login-section__content {
    width: 100%;
    margin: 20px;

    h3 {
      text-align: center;
      font-size: 1.4rem;
    }

    .input-section {
      width: 100%;
      display: flex;
      justify-content: center;
      border: 2px solid $black-color;
      border-radius: 5px;
      font-size: 1.4rem;

      input {
        border: none;
        width: 100%;
        padding: 10px 20px;
        font-size: 1.4rem;
        border-radius: 5px;
        background: $input-color;
        outline: none;
      }

      input:focus {
        background: $white-color;
      }

      input:focus::-webkit-input-placeholder {
        color: transparent;
      }

      input:focus:-moz-placeholder {
        color: transparent;
      }

      input:focus::-moz-placeholder {
        color: transparent;
      }

      input:focus:-ms-input-placeholder {
        color: transparent;
      }

      button {
        background: $black-color;
        border: none;
        color: $white-color;
        text-transform: uppercase;
        padding: 10px 20px;
        outline: none;
        cursor: pointer;
        border-left: 2px solid $black-color;
      }

      button:hover {
        font-weight: 700;
        background: $white-color;
        color: $black-color;
        border-radius: 0 5px 5px 0;
      }
    }
  }
}
