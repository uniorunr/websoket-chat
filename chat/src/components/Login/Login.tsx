import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import * as actions from '../../actions/actions';
import './Login.scss';

class Login extends Component<LoginProps> {
  private inputRef: React.RefObject<HTMLInputElement> = React.createRef();

  componentDidMount(): void {
    document.addEventListener('keydown', this.keyboardListener);
  }

  componentWillUnmount(): void {
    document.removeEventListener('keydown', this.keyboardListener);
  }

  keyboardListener = (event: KeyboardEvent) => {
    if (
      event.code === 'Enter' &&
      document.activeElement === this.inputRef.current &&
      this.inputRef.current &&
      this.inputRef.current.value
    ) {
      this.handleUsernameInput();
    }
  };

  handleUsernameInput = () => {
    const { setUserName, updateAuthStatus } = this.props;
    if (this.inputRef.current && this.inputRef.current.value) {
      const userName = this.inputRef.current.value;
      setUserName(userName);
      localStorage.setItem('userName', userName);
      updateAuthStatus(true);
    }
  };

  render() {
    return (
      <section className="login-section">
        <div className="login-section__content">
          <h3>Please tell us your name, buddy!</h3>
          <div className="input-section">
            <input placeholder="Your name..." ref={this.inputRef} />
            <button type="button" onClick={this.handleUsernameInput}>
              submit
            </button>
          </div>
        </div>
      </section>
    );
  }
}

interface LoginProps {
  setUserName: Function;
  updateAuthStatus: Function;
}

const mapDispatchToProps = (dispatch: Dispatch) => {
  const { setUserName, updateAuthStatus } = bindActionCreators(
    actions,
    dispatch
  );
  return {
    setUserName: (name: string) => {
      setUserName(name);
    },
    updateAuthStatus: (status: boolean) => {
      updateAuthStatus(status);
    }
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(Login);
