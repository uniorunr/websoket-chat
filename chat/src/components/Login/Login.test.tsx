import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../store/mainStore';
import Login from './Login';

it('Login component renders correctly', () => {
  const app = shallow(
      <Provider store={store}>
        <Login />
      </Provider>
  );
  expect(app).toMatchSnapshot();
});
