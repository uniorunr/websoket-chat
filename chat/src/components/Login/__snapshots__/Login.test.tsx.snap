// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<section
  className="login-section"
>
  <div
    className="login-section__content"
  >
    <h3>
      Please tell us your name, buddy!
    </h3>
    <div
      className="input-section"
    >
      <input
        placeholder="Your name..."
      />
      <button
        onClick={[Function]}
        type="button"
      >
        submit
      </button>
    </div>
  </div>
</section>
`;
